socs/agents/acu/agent.py

```diff
@@ -1377,12 +1398,135 @@ def _run_track(self, session, point_gen, step_time, azonly=False,
+
+        target_instance_id = self.agent.agent_address.split('.')[-1]
+        exercisor.set_client(target_instance_id)
+        settings = super_plan.get('settings', {})
+
+        plan_idx = 0
+        plan_t = None
+
+        for plan in super_plan['steps']:
+            plan['iter'] = iter(plan['driver'])
+
+        while session.status in ['running']:
+            time.sleep(1)
+            session.data['timestamp'] = time.time()
+            session.data['iterations'] += 1
+
+            # Fault maintenance
+            faults = exercisor.get_faults()
+            if faults['safe_lock']:
+                self.log.info('SAFE lock detected, exiting')
+                return _exit_now(False, 'Exiting on SAFE lock.')
+
+            if faults['local_mode']:
+                self.log.info('LOCAL mode detected, exiting')
+                return _exit_now(False, 'Exiting on LOCAL mode.')
+
+            if faults['az_summary']:
+                if session.data['attempts'] > 5:
+                    self.log.info('Too many az summary faults, exiting.')
+                    return _exit_now(False, 'Too many az summary faults.')
+                session.data['attempts'] += 1
+                self.log.info('az summary fault -- trying to clear.')
+                exercisor.clear_faults()
+                time.sleep(10)
+                continue
+
+            session.data['attempts'] = 0
+
+            # Plan execution
+            active_plan = super_plan['steps'][plan_idx]
+            if plan_t is None:
+                plan_t = time.time()
+
+            now = time.time()
+            if now - plan_t > active_plan['duration']:
+                plan_idx = (plan_idx + 1) % len(super_plan['steps'])
+                plan_t = None
+                continue
+
+            if settings.get('use_boresight'):
+                bore_target = random.choice(settings['boresight_opts'])
+                self.log.info(f'Setting boresight={bore_target}...')
+                _publish_activity('boresight')
+                exercisor.set_boresight(bore_target)
+
+            plan, info = next(active_plan['iter'])
+
+            self.log.info(f'Launching next scan. plan={plan}')
+
+            _publish_activity(active_plan['driver'].code)
+            ok = None
+            if 'targets' in plan:
+                exercisor.steps(**plan)
+            else:
+                exercisor.scan(**plan)
+            _publish_activity('idle')
+
+            if ok is None:
+                self.log.info('Scan completed without error.')
+            else:
+                self.log.info(f'Scan exited with error: {ok}')
+                _publish_error()
+
+        return _exit_now(True, "Stopped run process")
+
 
 def add_agent_args(parser_in=None):
     if parser_in is None:
         parser_in = argparse.ArgumentParser()
     pgroup = parser_in.add_argument_group('Agent Options')
     pgroup.add_argument("--acu_config")
+    pgroup.add_argument("--exercise_plan")
```

Comment:
The exercise plan arg is documented in the agent reference page as `--exercise-plan` with a hyphen. I see an underscore is probably used for consistency with `--acu_config`, but I think that should also use a hyphen for consistency across all of socs.

I'm fine with however you want to handle this at the moment, but should be consistent within this PR.